File: uri/ad-hock/grandma_day/1591.cpp
#include <iostream>
#include <string>
#include <vector>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    
    int t;
    cin >> t;
    
    while (t--) {
        int l, c;
        cin >> l >> c;
        
        vector<string> grid(l);
        for (int i = 0; i < l; i++) {
            cin >> grid[i];
        }
        
        int p;
        cin >> p;
        
        while (p--) {
            string word;
            cin >> word;
            int len = word.length();
            int count = 0;
            
            if (len == 1) {
                for (int i = 0; i < l; i++) {
                    for (int j = 0; j < c; j++) {
                        if (grid[i][j] == word[0]) {
                            count++;
                        }
                    }
                }
            } else {
                for (int i = 0; i < l; i++) {
                    for (int j = 0; j <= c - len; j++) {
                        bool match = true;
                        for (int k = 0; k < len && match; k++) {
                            if (grid[i][j + k] != word[k]) match = false;
                        }
                        if (match) count++;
                    }
                }
                
                for (int j = 0; j < c; j++) {
                    for (int i = 0; i <= l - len; i++) {
                        bool match = true;
                        for (int k = 0; k < len && match; k++) {
                            if (grid[i + k][j] != word[k]) match = false;
                        }
                        if (match) count++;
                    }
                }
            }
            
            cout << count << '\n';
        }
    }
    
    return 0;
}

File: uri/ad-hock/grandma_day/1591.rs
use std::io::{self, BufRead, Write, BufWriter};

fn main() {
    let stdin = io::stdin();
    let stdout = io::stdout();
    let mut out = BufWriter::new(stdout.lock());
    let mut lines = stdin.lock().lines();
    
    let t: i32 = lines.next().unwrap().unwrap().trim().parse().unwrap();
    
    for _ in 0..t {
        let line = lines.next().unwrap().unwrap();
        let mut iter = line.split_whitespace();
        let l: usize = iter.next().unwrap().parse().unwrap();
        let c: usize = iter.next().unwrap().parse().unwrap();
        
        let mut grid: Vec<Vec<char>> = Vec::with_capacity(l);
        for _ in 0..l {
            let row: Vec<char> = lines.next().unwrap().unwrap().trim().chars().collect();
            grid.push(row);
        }
        
        let p: i32 = lines.next().unwrap().unwrap().trim().parse().unwrap();
        
        for _ in 0..p {
            let word: Vec<char> = lines.next().unwrap().unwrap().trim().chars().collect();
            let len = word.len();
            let mut count = 0;
            
            if len == 1 {
                for i in 0..l {
                    for j in 0..c {
                        if grid[i][j] == word[0] {
                            count += 1;
                        }
                    }
                }
            } else {
                for i in 0..l {
                    for j in 0..=c.saturating_sub(len) {
                        if j + len <= c {
                            let mut matched = true;
                            for k in 0..len {
                                if grid[i][j + k] != word[k] {
                                    matched = false;
                                    break;
                                }
                            }
                            if matched { count += 1; }
                        }
                    }
                }
                
                for j in 0..c {
                    for i in 0..=l.saturating_sub(len) {
                        if i + len <= l {
                            let mut matched = true;
                            for k in 0..len {
                                if grid[i + k][j] != word[k] {
                                    matched = false;
                                    break;
                                }
                            }
                            if matched { count += 1; }
                        }
                    }
                }
            }
            
            writeln!(out, "{}", count).unwrap();
        }
    }
}
